src/main/java/teammates/ui/pagedata/AdminActivityLogPageData.java

```diff
@@ -148,99 +157,75 @@ public void generateQueryParameters(String query) {
     }
 
     /**
-     * check current log entry should be excluded as rubbish logs
-     * returns false if the logEntry is regarded as rubbish
+     * Checks whether the current log entry should be included or not.
+     *
+     * @returns true if it should be included
      */
-    private boolean shouldExcludeLogEntry(ActivityLogEntry logEntry) {
-
+    private boolean shouldIncludeLogEntry(ActivityLogEntry logEntry) {
         if (ifShowAll) {
-            return false;
+            return true;
         }
 
         for (String uri : excludedLogRequestURIs) {
 
-            if (uri.contains(logEntry.getServletName())) {
-                return true;
+            if (uri.contains(logEntry.getUrl())) {
+                return false;
             }
         }
 
-        return false;
+        return true;
     }
 
     /**
      * Performs the actual filtering, based on QueryParameters
-     * returns false if the logEntry fails the filtering process
+     * returns false if the logEntry fails the filtering process.
+     * <p>
+     * If the queryMessage is null, the function will return true
+     * and set the queryMessage with error message.
      */
-    public ActivityLogEntry filterLogs(ActivityLogEntry logEntry) {
-        if (!logEntry.toShow()) {
-            return logEntry;
-        }
-
+    public boolean filterLogs(ActivityLogEntry logEntry) {
```

Comment:
Yes. The code quality of query process can be improved a lot. So I have written test cases in #6655, after that I can begin to refactor it.